### ISSUE TITLE
Use self-explanatory, meaningful argument names for loss functions, regularizers and models

### ISSUE THREAD

**Issue by lukostaz** [Accenture/AmpliGraph#31]
**Background and Context**
The signatures of all functions and classes of the library should be easily understandable. Argument names should be intuitive, and default values should lead to viable results, even without user-defined preferences.

**Description**
Currently, `loss_functions.py` and `regularizers.py` inherited classes such as `PairwiseLoss` use the `hyperparam_dict` argument in their `__init__()` method. Such argument should be replaced with Python's `kwargs` approach, and unpakced into a list of meaningful argument names (e.g. `margin`, `gamma`).
